hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/tools/federation/RouterAdmin.java

```diff
@@ -462,6 +484,142 @@ public int run(String[] argv) throws Exception {
     return exitCode;
   }
 
+  /**
+   * Add all mount point entries provided in the request.
+   *
+   * @param parameters Parameters for the mount points.
+   * @param i Current index on the parameters array.
+   * @return True if adding all mount points was successful, False otherwise.
+   * @throws IOException If the RPC call to add the mount points fail.
+   */
+  private boolean addAllMount(String[] parameters, int i) throws IOException {
+    List<AddMountAttributes> addMountAttributesList = new ArrayList<>();
+    Set<String> mounts = new HashSet<>();
+    while (i < parameters.length) {
+      AddMountAttributes addMountAttributes = getAddMountAttributes(parameters, i, true);
+      if (addMountAttributes == null) {
+        return false;
+      }
+      if (mounts.contains(addMountAttributes.getMount())) {
+        System.err.println("Multiple inputs for mount: " + addMountAttributes.getMount());
+        return false;
+      }
+      mounts.add(addMountAttributes.getMount());
```

Comment:
HashSet add method returns false, if the entry was already there, contains ain't required, unless I am missing something.

btw. earlier this was a success case for you, the later entry used to take precedence. I tried it....

Reply:
> btw. earlier this was a success case for you, the later entry used to take precedence. I tried it....

i forgot to reply to this yesterday: yes this was success case and only later entry was taking precedence before this addendum, that is due to the nature of state store putAll implementation. 
but good for us to prevent such inputs with the proper error message so that user can rectify it. that way, we won't have to rely on state store impl to resolve it with different behavior (i.e. guard against any behavior changes of putAll impl in all state stores).